rts/c/cuda.h

```diff
@@ -249,6 +257,45 @@ static int cuda_device_setup(struct cuda_context *ctx) {
     fprintf(stderr, "Using device #%d\n", chosen);
   }
 
+  int used_devices = 0;
+  int* devices = (int*)malloc(sizeof(int)*count);
+  for(int devID = 0; devID < count; devID++){
+    CUDA_SUCCEED_FATAL(cuDeviceGet(&dev, devID)); 
+
+    int cc_major = device_query(dev, COMPUTE_CAPABILITY_MAJOR);
+    int cc_minor = device_query(dev, COMPUTE_CAPABILITY_MINOR);
+    // Here it's a design chocie that all device should be of the same CC
+    if (cc_major == cc_major_best && cc_minor == cc_minor_best){
+      devices[used_devices] = devID;
+      used_devices++;
+    }
+  }
+  if(ctx->cfg.debugging){
+    fprintf(stderr, "Using %d devices\n", used_devices);
+  }
+  ctx->kernel_iterator = 0;
+  ctx->device_count = used_devices;
+  ctx->devices = (CUdevice*)malloc(sizeof(CUdevice)*used_devices);
```

Comment:
I prefer `calloc()` for this.  Yes, I know the rest of the code isn't consistent.  Be better than I was!